quinn/src/endpoint.rs

```diff
@@ -441,6 +437,15 @@ pub(crate) struct Shared {
 }
 
 impl State {
+    fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
+        self.recv_state.recv_limiter.start_cycle();
+        let poll_res = self
```

Comment:
Maybe add a comment about why it's correct to skip `finish_cycle()` on error returns of `poll_socket()`?

Reply:
It doesn't matter either way because errors here are fatal to the endpoint driver. I refactored to not skip it because that was more concise than a comment explaining why we can.